Manual.md

```diff
@@ -0,0 +1,145 @@
+	- During this stage `crew` `cd`'s into the extracted directory, and stays there until the `preinstall` stage 
+4. `self.patch` - Only `patch` commands should be used during this stage	- If provided
+5. `self.build` - Package is compiled
+6. `self.check` - Package is then checked - If provided
+7. `self.preinstall` - Pre-install checks/edits are done - If provided 
+	- `crew` now `cd`'s out of the directory provided by the package
+8. `self.install` - Commands to install to `#{CREW_DEST_DIR}` are run
+	- Only sourced when the package is being built
+		- `self.install` is not required to be defined
+			- Install may be carried out during `self.build` or `self.preinstall`
+9. `self.postinstall` - Post-install checks/edits are done - If provided
+10. Repack 
+	 - Only binary packages call this stage 
+		 - Contents are checked and repackaged then installed to `#{CREW_PREFIX}`
+***
+## Binary
+Binary packages follow a different set of rules then source packages.
+Only the functions `self.preinstall` and `self.postinstall` are run during a binary package install.
+`crew` requires a few things to exist in the archive in order to install a binary package.
+
+- `dlist` a list of directories used/installed to by the package
+- `filelist` a list of all files included in the package
+***
+## Variables 
+
+The following variables *must*/should be set by a package script. - [Rundown](#rundown)
+
+- `class # < Package` - The`#` must be name of the name of your package	. It must start with a capital and be the same as the filename. (`#.rb`)
+- `description`  -  Must be set. Is the description of the package.
+- `homepage`  - Must be set. Is the homepage of the package.
+- `version`  - Must be set. Is the package version.
+- `compatibility`  - Which architectures the package can operate on.
+- `source_url`  Is the URL where the source package can be found.<sup>[More](#More)</sup>
+- `source_sha256`  - The checksum for the package which will be downloaded from the `source_url`.
+> ### Environment Variables
+
+The following variables are preset by `crew`
+
+- `#{CREW_PREFIX}` - The prefix used by `crew` - Equal to `/usr/local`
+- `#{CREW_LIB_PREFIX}` - The `LIB` prefix used by `crew` - Equal to `#{CREW_PREFIX}/lib`
+- `#{CREW_DEST_DIR}` - The`DESTDIR` variable used by `crew` - Equal to `#{CREW_PREFIX}/tmp/crew/dest`
+- `#{CREW_DEST_PREFIX}` - The`DESTDIR` variable prefix used by `crew` - Equal to `#{CREW_DEST_DIR}/usr/local`
+- `#{CREW_OPTIONS}` - The preset options for `./congifure` - <sup>[Equals](#=)</sup>
+- `#{CREW_MAN_PREFIX}` - Useful for building man pages - Equal to `#{CREW_PREFIX}/share/man`
+- `#{CREW_DEST_LIB_PREFIX}` - The `DESTDIR` `LIB` prefix - Equal to `#{CREW_DEST_PREFIX}/lib`
+- `#{CREW_DEST_HOME}` - The `DESTDIR` home variable - Equal to `#{CREW_DEST}/#{HOME}}`
+- `#{HOME}` - Variable used by `crew` for `$HOME`
+- `#{ARCH}` - Variable used by `crew` for `$(arch)`
```

Comment:
Might want to mention these constants can be displayed with the `crew const` command.